sr_robot_commander/test/test_robot_commander.py

```diff
@@ -0,0 +1,676 @@
+
+import collections
+from builtins import round
+import copy
+import rospy
+import rostest
+from unittest import TestCase
+from sr_robot_commander.sr_robot_commander import SrRobotCommander, MoveGroupCommander, PlanningSceneInterface
+from geometry_msgs.msg import Pose, PoseStamped
+from control_msgs.msg import FollowJointTrajectoryActionGoal
+from moveit_msgs.msg import RobotState, RobotTrajectory
+from moveit_msgs.srv import GetPositionFK, SaveRobotStateToWarehouse
+from trajectory_msgs.msg import JointTrajectoryPoint, JointTrajectory
+from std_msgs.msg import Header
+from sensor_msgs.msg import JointState
+from moveit_commander.exception import MoveItCommanderException
+from moveit_commander import conversions
+from actionlib_msgs.msg import GoalStatusArray
+import tf2_ros
+import time
+
+PKG = "sr_robot_commander"
+RA_HOME_ANGLES = {'ra_shoulder_pan_joint': 0.00, 'ra_elbow_joint': 2.00,
+                  'ra_shoulder_lift_joint': -1.57, 'ra_wrist_1_joint': -0.73,
+                  'ra_wrist_2_joint': 1.57, 'ra_wrist_3_joint': 0.00}
+
+RA_EXAMPLE_TARGET = {'ra_shoulder_pan_joint': 0.2, 'ra_elbow_joint': 1.80,
+                     'ra_shoulder_lift_joint': -1.37, 'ra_wrist_1_joint': -0.52,
+                     'ra_wrist_2_joint': 1.57, 'ra_wrist_3_joint': 0.00}
+
+
+class TestSrRobotCommander(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        rospy.wait_for_message("/move_group/status", GoalStatusArray)
+        cls.robot_commander = SrRobotCommander("right_arm")
+        cls.robot_commander.set_planner_id("RRTstarkConfigDefault")
+        cls.robot_commander.set_planning_time(3)
+        cls.eef = cls.robot_commander.get_end_effector_link()
+
+    def reset_to_home(self):
+        self.robot_commander._reset_plan()
+        plan = self.robot_commander.plan_to_joint_value_target(RA_HOME_ANGLES, angle_degrees=False,
+                                                               custom_start_state=None)
+        self.robot_commander.execute_plan(plan)
+
+    def compare_poses(self, pose1, pose2, tolerance=0.01):
+        p1_list = [pose1.position.x, pose1.position.y, pose1.position.z,
+                   pose1.orientation.x, pose1.orientation.y, pose1.orientation.z, pose1.orientation.w]
+        p2_list = [pose2.position.x, pose2.position.y, pose2.position.z,
+                   pose2.orientation.x, pose2.orientation.y, pose2.orientation.z, pose2.orientation.w]
+        digit = 2
+        p1_list = [round(i, digit) for i in p1_list]
+        p2_list = [round(i, digit) for i in p2_list]
+
+        for idx, vals in enumerate(zip(p1_list, p2_list)):
+            if vals[0] == -0.0:
+                p1_list[idx] = 0
+            if vals[1] == -0.0:
+                p2_list[idx] = 0
+            if abs(p1_list[idx]) - abs(p2_list[idx]) > tolerance:
+                return False
+        return True
+
+    def compare_joint_states(self, js1, js2, tolerance=0.01):
+        common_keys = set(js1.keys()).intersection(set(js2.keys()))
```

Comment:
Changed keys to joint_states. The intersection() part is to get the common keys between 2 joint state - this can be used i.e we would like to compare a joint_state which has 6 joints with another joints_state where there are only 2 joints.